flytekit/clients/friendly_rs.py

```diff
@@ -0,0 +1,166 @@
+from flyteidl.admin import common_pb2 as _common_pb2
+from flyteidl.admin import task_pb2 as _task_pb2
+
+import flyrs
+from flytekit.clients.friendly import SynchronousFlyteClient as _SynchronousFlyteClient
+from flytekit.configuration import PlatformConfig
+from flytekit.models import common as _common
+from flytekit.models import filters as _filters
+from flytekit.models import task as _task
+from flytekit.models.core import identifier as _identifier
+
+
+class RustSynchronousFlyteClient(_SynchronousFlyteClient):
```

Comment:
Can we a disclaimer that this currently only works with unauthenticated requests?

Also, we should discuss how Auth is going to be implemented (essentially reflect [this](https://github.com/flyteorg/flytekit/blob/e3f92cf748c7852164881bc566481dfe5b3d3e40/flytekit/clients/raw.py#L53-L58) logic in rust).

Reply:
Of course, thanks for reminding.

Plan to work on `Auth` and `Rust gRPC channel interceptor` in subsequent PRs.

I'd love to discuss how `Auth` is going to be implemented and the ability to test all the various `Oauth` setups.
I'm totally a newbie in authentication mechanisms.